### PR TITLE
adds numerous keepass plugins

### PR DESCRIPTION
**Pull request by nazarewk**
###### Motivation for this change

Adds support for:
- [x] de-facto (as in built-into Keepass2Android etc.) standard plugin for handling OTPs in KeePass
- [x] copying passwords & fields on character-by-character basis without using autotype
- [ ] wayland autopaste enabler
- [x] plugin displaying fields as QRCodes for easy transfer between devices

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Tested by incorporating [into my NixOS configuration](https://github.com/nazarewk-iac/nix-configs/blob/a8f3b60ce5dafaa90e0afb035fe07bca15db9bd1/flake.nix#L30-L32)

### CODE REVIEW

File: pkgs/applications/misc/keepass-plugins/qrcodeview/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  drv = stdenv.mkDerivation {
+    inherit pname version;
+
+    src = fetchurl {
+      url = "https://github.com/JanisEst/KeePassQRCodeView/releases/download/v${version}/KeePassQRCodeView.plgx";
+      sha256 = "e13c9f02bb0d79b479ca0e92490b822b5b88f49bb18ba2954d3bbe0808f46e6d";
+    };
+
+    meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/keepass-plugins/qrcodeview/default.nix:15]: meta should be last.

File: pkgs/applications/misc/keepass-plugins/qrcodeview/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{ lib, stdenv, buildEnv, fetchurl, mono }:
+
+let
+  pname = "keepass-qrcodeview";
+  version = "1.0.4";
+
+  drv = stdenv.mkDerivation {
+    inherit pname version;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/keepass-plugins/qrcodeview/default.nix:8]:
```suggestion
  drv = stdenv.mkDerivation {
    pname = "keepass-qrcodeview";
    version = "1.0.4";
```

**Review by SuperSandro2000 — COMMENTED** [review]
The descriptions are a bit long for description and have more the length of longDescription. Also it would be nice if they could have a new line or two to fit onto one screen.

Also please follow the contributing guide when creating and naming your commits.

**Comment by nazarewk** [comment]
@SuperSandro2000 i've cleaned up the commit history and applied your suggestions, last thing I am not sure about is whether i should add any changelog/release entries myself?

**Comment by SuperSandro2000** [comment]
> last thing I am not sure about is whether i should add any changelog/release entries myself?

Probably not necessary. It isn't a breaking change and not major enough for everyone to read.

**Pull request merged**